### PR TITLE
Elena burak/exception middleware

### PR DESCRIPTION
**Pull request by ElenaBurak**

### CODE REVIEW

File: InternshipBE/ExaLearn.Shared/Extensions/GlobalExceptionMiddlewareExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using ExaLearn.Shared.Middleware;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Logging;
+
+
```
Comment (romanstat) [inline InternshipBE/ExaLearn.Shared/Extensions/GlobalExceptionMiddlewareExtensions.cs:3]: remove empty line

File: InternshipBE/ExaLearn.Shared/Extensions/GlobalExceptionMiddlewareExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using ExaLearn.Shared.Middleware;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Logging;
+
+namespace ExaLearn.Shared.Extensions
+{
+    public static class GlobalExceptionMiddlewareExtensions
+    {
+        public static void UseGlobalExceptionMiddleware(this IApplicationBuilder app, ILogger logger)
```
Comment (romanstat) [inline InternshipBE/ExaLearn.Shared/Extensions/GlobalExceptionMiddlewareExtensions.cs]: remove logger

File: InternshipBE/ExaLearn.Shared/Middleware/ExceptionMiddleware.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+using Microsoft.Extensions.Logging;
+
+namespace ExaLearn.Shared.Middleware
+{
+    public class ExceptionMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ILogger _logger;
+
+        public ExceptionMiddleware(RequestDelegate next, ILogger logger)
```
Comment (romanstat) [inline InternshipBE/ExaLearn.Shared/Middleware/ExceptionMiddleware.cs]: change logger DI

File: InternshipBE/ExaLearn.WebApi/Startup.cs [hunk 4]
```diff
@@ -36,7 +38,7 @@ public void ConfigureServices(IServiceCollection services)
             services.AddDbContext<ExaLearnDbContext>(options => options.UseNpgsql(Configuration.GetConnectionString("DbContext")));
         }
 
-        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env, ILogger logger)
```
Comment (romanstat) [inline InternshipBE/ExaLearn.WebApi/Startup.cs]: remove logger

File: InternshipBE/ExaLearn.Shared/Middleware/ExceptionMiddleware.cs (hunk 3 above)
Comment (ElenaBurak) [reply InternshipBE/ExaLearn.Shared/Middleware/ExceptionMiddleware.cs]: changed

**Review by romanstat — APPROVED** [review]

**Review by romanstat — APPROVED** [review]

**Review by ExadelSandbox — APPROVED** [review]

**Pull request merged**